### DISCUSSION TITLE
How do I mass-mail my users?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#3316] [Q&A] [answered]
Curious to know if anything is planned for sending out system announcements to `select * from users where verified=1` :)

**Comment by ganigeorgiev** [comment, marked as answer]
There is no such builtin functionality at the moment (_and I'm not also sure if it can be generilized_), but you can access the app mailer and create it on your own.

Here is an example using the JS `pb_hooks` and a custom console command (_I haven't tested it; the same functionality could be attached to a hook, custom route, cron job, etc._):

```js
// pb_hooks/main.pb.js

// registers `./pocketbase massmail` command
$app.rootCmd.addCommand(new Command({
    use: "massmail",
    run: (cmd, args) => {
        const users = arrayOf(new Record);
        $app.dao().recordQuery("users")
            .andWhere($dbx.exp("verified = 1 AND email != ''"))
            .all(users)

        const mailClient = $app.newMailClient()

        for (const user of users) {
            const email = user.email()

            mailClient.send(new MailerMessage({
                from: {
                    address: $app.settings().meta.senderAddress,
                    name:    $app.settings().meta.senderName,
                },
                to:      [{address: email}],
                subject: "YOUR_SUBJECT...",
                html:    "YOUR_HTML_BODY...",
                // plain text version is autogenerated if not explicitly set
                // bcc, cc and custom headers are also supported...
            }))

            console.log("Successfully sent email to " + email)
        }
    },
}))
```

_Used apis:_
- https://pocketbase.io/docs/js-console-commands/
- https://pocketbase.io/docs/js-sending-emails/
- https://pocketbase.io/docs/js-records/#custom-record-query

_If you want the email to be constructed from an HTML template, you can also check the [$template helpers](https://pocketbase.io/docs/js-rendering-templates/#overview)._

**Reply by benallfree** [reply]
Another +1 for an admin plugin architecture :)
